# Case Study on Measurement of Central Tendency and Dispersion

In [136]:
#importing important libraries
import pandas as pd
import numpy as np
import matplotlib.pyplot as plt
import seaborn as sns

In [137]:
#Reading dataset to python environment
data=pd.read_csv('StudentsPerformance.csv')

In [138]:
data.head()

gender race/ethnicity parental level of education         lunch  \
0  female        group C            some high school  free/reduced   
1  female        group B                 high school  free/reduced   
2  female        group B            some high school  free/reduced   
3  female        group B                some college      standard   
4  female        group C                some college  free/reduced   

  test preparation course  math score  reading score  writing score  
0                    none           0             17             10  
1                    none           8             24             23  
2                    none          18             32             28  
3                    none          11             38             32  
4                    none          22             39             33

In [162]:
#Checking for null values
data.isna().sum()

gender                         0
race/ethnicity                 0
parental level of education    0
lunch                          0
test preparation course        0
math score                     0
reading score                  0
writing score                  0
dtype: int64

# 1. Find out how many males and females participated in the test.

In [139]:
data.groupby('gender')['gender'].count()

gender
female    518
male      482
Name: gender, dtype: int64

Total number of Females participated in the test : 518
 And Total number of Males participated in the test : 482   

# 2. What do you think about the students' parental level of education?

In [140]:
data['parental level of education'].mode()

0    some college
dtype: object

The parental level of education for more number of students are 'some college'

# 3. Who scores the most on average for math, reading and writing based on


### 1) Gender


In [141]:
df1=data.groupby('gender')['math score'].mean().sort_values()
df1

gender
female    63.376448
male      68.821577
Name: math score, dtype: float64

Male students scores the most on average for math.

In [142]:
df2=data.groupby('gender')['reading score'].mean().sort_values()
df2

gender
male      65.545643
female    72.590734
Name: reading score, dtype: float64

Females scores the most on average for reading

In [143]:
df3=data.groupby('gender')['writing score'].mean().sort_values()
df3

gender
male      63.446058
female    72.467181
Name: writing score, dtype: float64

Females scores the most on average for writing

From the above result we can conclude that the female students scores more in reading and writing comparing to the male students. Male students scores more in math than the female students.

### 2) Test preparation course

In [144]:
d1=data.groupby('test preparation course')['math score'].mean().sort_values()
d1

test preparation course
none         63.78972
completed    69.96648
Name: math score, dtype: float64

The students who completed the Test preparation course scores more on average for math

In [145]:
d2=data.groupby('test preparation course')['reading score'].mean().sort_values()
d2

test preparation course
none         66.417445
completed    74.175978
Name: reading score, dtype: float64

The students who completed the Test preparation course scores more on average for reading

In [146]:
d3=data.groupby('test preparation course')['writing score'].mean().sort_values()
d3

test preparation course
none         64.457944
completed    74.684358
Name: writing score, dtype: float64

The students who completed the Test preparation course scores more on average for writing

From the above result we can conclude that the students who have completed the Test preparation course scores more in math, reading and writing comparing to the students those who haven't.

# 4. What do you think about the scoring variation for math, reading and writing based on

### 1) Gender

In [147]:
d_var1=data.groupby('gender')['math score'].var().sort_values()
d_var1

gender
male      211.889097
female    256.958593
Name: math score, dtype: float64

In [148]:
d_var2=data.groupby('gender')['reading score'].var().sort_values()
d_var2

gender
male      200.211010
female    207.677438
Name: reading score, dtype: float64

In [149]:
d_var3=data.groupby('gender')['writing score'].var().sort_values()
d_var3

gender
male      202.413924
female    220.369327
Name: writing score, dtype: float64

Here male students have lesser variance.

### 2)  Test preparation course

In [150]:
d_var4=data.groupby('test preparation course')['math score'].var().sort_values()
d_var4

test preparation course
completed    210.884027
none         246.668662
Name: math score, dtype: float64

In [151]:
d_var5=data.groupby('test preparation course')['reading score'].var().sort_values()
d_var5

test preparation course
completed    183.265864
none         213.419851
Name: reading score, dtype: float64

In [152]:
d_var6=data.groupby('test preparation course')['writing score'].var().sort_values()
d_var6

test preparation course
completed    175.202612
none         226.251739
Name: writing score, dtype: float64

A small variance indicates that the data points tend to be very close to the mean, and to each other. A high variance indicates that the data points are very spread out from the mean, and from one another. Here for the students completed the Test preparation course have lesser variance. That means they are closer to the mean value.  

# 5. The management needs your help to give bonus points to the top 25% of students based on their maths score, so how will you help the management to achieve this.


In [153]:
dq= data['math score'].quantile(q=0.75)
dq

77.0

In [154]:
data['math score'] >= 77

0      False
1      False
2      False
3      False
4      False
       ...  
995     True
996     True
997     True
998     True
999     True
Name: math score, Length: 1000, dtype: bool

In [155]:
data_math=data[data['math score'].ge(dq)].sort_values(by='math score', ascending=False)
data_math

gender race/ethnicity parental level of education         lunch  \
999    male        group E           bachelor's degree      standard   
998    male        group E          associate's degree  free/reduced   
997    male        group D                some college      standard   
996    male        group A                some college      standard   
517  female        group E          associate's degree      standard   
..      ...            ...                         ...           ...   
854    male        group D            some high school      standard   
855    male        group E          associate's degree  free/reduced   
856    male        group E            some high school      standard   
415  female        group B             master's degree  free/reduced   
414  female        group A                 high school  free/reduced   

    test preparation course  math score  reading score  writing score  
999               completed         100            100            100  
998               completed         100            100             93  
997               completed         100             97             99  
996               completed         100             96             86  
517                    none         100            100            100  
..                      ...         ...            ...            ...  
854               completed          77             68             69  
855               completed          77             69             68  
856               completed          77             76             77  
415               completed          77             97             94  
414               completed          77             88             85  

[255 rows x 8 columns]

In [157]:
data_math.shape

(255, 8)

From the above result, we can find out that the top 25% includes 255 students who are eligible for the bonus points.